File: CS50/Week-2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int count_letters(string text);
int count_words(string Text);
int count_sentences(string Text);
int main(void)
{
    char Text[1000];
    string input = get_string ("Text: ");
    strcpy(Text, input);


    int L = count_letters(Text);
    int W = count_words(Text);
    int S = count_sentences(Text);

    float index = 0.0588 * (100 * (float)L / (float)W) - 0.296 * (100 * (float)S / (float)W) - 15.8;
    if(index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if(index <= 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %.0f\n", index);
    }

}
int count_letters(string Text)
{
    int Letters = 0;
    int i;

    for(i = 0; i < strlen(Text); i++)
    {
        if(isalpha(Text[i]))
        {
            Letters++;
        }
    }
    return Letters;
}
int count_words(string Text)
{
    int words = 0;
    int j;
    for(j = 0; j < strlen(Text); j++)
    if(isspace(Text[j]))
    {
        words++;
    }
    return words + 1;
}
int count_sentences(string Text)
{
    int sentences = 0;
    int k;
    for(k = 0; k < strlen(Text); k++)
    {
      if(Text[k] == '!' ||Text[k] == '.' || Text[k] == '?')
      {
        sentences++;
      }
    }
    return sentences;
}

File: CS50/Week-6/readability.py
# TODO

import cs50
import sys

def main():
    Text = cs50.get_string("Enter the text: ")

    L = count_letters(Text)
    W = count_words(Text)
    S = count_sentences(Text)

    index = 0.0588 * (100 * L / W) - 0.296 * (100 * S / W) - 15.8

    if index >= 16:
        print("Grade 16+")

    elif index <= 1:
        print("Before Grade 1")
    else:
        print(f"Grade {round(index)}")

def count_letters(Text):
    # Initialize letters to 0
    letters = 0

    for letter in Text:
        if letter.isalpha():
            letters += 1
    return letters

def count_words(Text):
    # Split the text into words and return the number of words
    words = len(Text.split())

    return words


def count_sentences(Text):
    # Initialize sentence
    sentences = 0

    for char in Text:
        if char in ["!", "?", "."]:
            sentences += 1
    return sentences

if __name__ == "__main__":
    main()
